spec/features/users/user_is_deactivated_spec.rb

```diff
@@ -0,0 +1,182 @@
+require 'rails_helper'
+
+feature 'user is deactivated' do
```

Comment:
Very robust testing here. I like how you cover all the edge cases for questions, comments, and answers. I also like how you test that deactivated users can still edit their profile, password, and view the page.